src/invoice.rs

```diff
@@ -208,6 +244,33 @@ impl FromStr for RgbInvoice {
                 }
             };
 
+        let mut query: IndexMap<String, String> = IndexMap::new();
+        if let Some(q) = uri.query() {
```

Comment:
I thought URI library should do that for us. Otherwise here we are also required to do URIDecode....

Reply:
decoding now handled by the lib's `decode()` method on `EStr` (on query parameter key/value separately, see also next requested change)